pkg/instance/start.go

```diff
@@ -131,20 +130,24 @@ func Prepare(ctx context.Context, inst *limatype.Instance) (*Prepared, error) {
 	}, nil
 }
 
-// Start starts the hostagent in the background, which in turn will start the instance.
-// Start will listen to hostagent events and log them to STDOUT until either the instance
+// StartWithPaths starts the hostagent in the background, which in turn will start the instance.
+// StartWithPaths will listen to hostagent events and log them to STDOUT until either the instance
 // is running, or has failed to start.
 //
-// The `limactl` argument allows the caller to specify the full path of the `limactl` executable.
-// When called from inside limactl itself it will always be the empty string which uses the name
-// of the current executable instead.
-//
-// The `launchHostAgentForeground` argument makes the hostagent run in the foreground.
+// The launchHostAgentForeground argument makes the hostagent run in the foreground.
 // The function will continue to listen and log hostagent events until the instance is
 // shut down again.
 //
-// Start calls Prepare by itself, so you do not need to call Prepare manually before calling Start.
-func Start(ctx context.Context, inst *limatype.Instance, limactl string, launchHostAgentForeground, showProgress bool) error {
+// The showProgress argument tells the hostagent to show provision script progress by tailing cloud-init logs.
+//
+// The limactl argument allows the caller to specify the full path of the limactl executable.
+// The guestAgent argument allows the caller to specify the full path of the guest agent executable.
+// Inside limactl this function is only called by Start, which passes empty strings for both
+// limactl and guestAgent, in which case the location of the current executable is used for
+// limactl and the guest agent is located from the corresponding <prefix>/share/lima directory.
+//
+// StartWithPaths calls Prepare by itself, so you do not need to call Prepare manually before calling Start.
+func StartWithPaths(ctx context.Context, inst *limatype.Instance, launchHostAgentForeground, showProgress bool, limactl, guestAgent string) error {
```

Comment:
The function seems to be called only from `Start()`? Then no need to make this function public?

Reply:
The whole point of having `limactl` and `guestAgent` parameters is for calling it externally (outside of limactl itself); internally it always passes empty strings to use the default mechanism to locate them.

So I could have just added `guestAgent` as an extra parameter to `Start`, but I found it silly to pass the empty strings everywhere, so I instead turned it into a wrapper.

Reply:
> What applications calls `Start` externally?

I have some; probably nobody else; idk. Currently playing with a Kubernetes controller that will provision Lima instances as an example of orchestration. Not yet open source, but will be, if it works out... I accept full responsibility of keeping it updated with whatever changes we make in Lima, but would like the capabilities to be exposed in some way.